src/index.ts

```diff
@@ -95,41 +96,26 @@ const calculateDepreciation = ({
     }]
   }
 
-  for (let index = 0; index < totalDepreciationYears; index++) {
-    // Current year
-    if (index === 0) {
-      const monthsLeftInFirstYear = MONTHS_IN_YEAR - purchaseMonth + 1;
-      const result = calculate(purchaseAmount, totalDepreciationYears, endAmount, monthsLeftInFirstYear);
-      results.push({
-        year: purchaseYear,
-        depreciationMonths: monthsLeftInFirstYear,
-        ...result,
-      });
-      endAmount = result.endAmount;
-
-      // If item is purchased in Jan, monthsLeftInLastYear is 0
-      // If item is purchased in Dec, monthsLeftInLastYear is 11
-      monthsLeftInLastYear = MONTHS_IN_YEAR - monthsLeftInFirstYear;
-    } else { // Future years
-      const result = calculate(purchaseAmount, totalDepreciationYears, endAmount);
-      results.push({
-        year: purchaseYear + index,
-        depreciationMonths: MONTHS_IN_YEAR,
-        ...result,
-      });
-      endAmount = result.endAmount;
-    }
-  }
+  const parts: number = purchaseMonth > 1 ? totalDepreciationYears + 1 : totalDepreciationYears;
+  const monthInFirstYear : number = MONTHS_IN_YEAR - purchaseMonth + 1;
+  const monthInLastYear : number = monthInFirstYear === MONTHS_IN_YEAR ? MONTHS_IN_YEAR : MONTHS_IN_YEAR - monthInFirstYear;
 
-  if (monthsLeftInLastYear > 0) {
+  let monthInThisYear : number = monthInFirstYear;
+  endAmount = purchaseAmount;
```

Comment:
this line can be removed, because in [line 86](https://github.com/kontist/asset-depreciation-calculator/pull/4/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80) `endAmount` initialized with `purchaseAmount`.